### ISSUE TITLE
Error reporting for unknown library verisons

### ISSUE THREAD

**Issue by pkch** [python/mypy#2898]
When type-checking a python 3.6 file that contains `from enum import auto`, the latest build of `mypy` ( commit 0bb2d16 from master) complains: `error: Module 'enum' has no attribute 'auto'`. The reason is that `build.py: State.__init__()` finds the stub for `enum` using `path = find_module(file_id, manager.lib_path)`; and the paths in `manager.lib_path` are something like

```('.../typeshed/stdlib/3.4', .../typeshed/stdlib/3.3', '.../typeshed/stdlib/3', '.../third_party/3', '.../stdlib/2and3', '.../third_party/2and3')```

It's understandable that stubs aren't yet available for the most recent library versions in `typeshed`, but I think `mypy` should perhaps warn the user about it to make it easier to figure out what the problem is. For instance, whenever a library stub used is for an older version, a warning could be logged?

Related to this is the fact that the `default.py: PYTHON3_version` is (3, 5); perhaps it should be updated to (3, 6)? As is, the `default_lib_path` doesn't even attempt to search in `.../typeshed/stdlib/3.6` folder. (This wouldn't have helped in my case, since there's no `enum.pyi` in the 3.6 folder.)

Also related is #2897 that I just opened.

**Comment by pkch** [comment]
I just realized that the `3.4/enum.py` actually does know about `auto` (which was introduced in python 3.6)! It just puts it behind an if-statement that checks the current version of python.

This means that there is no simple way to know whether a given library version is supported or not (in this case enum from 3.6 is supported, but it's not as easy to confirm as just check for the presence of the file `3.6/enum.pyi`).

The only problem in my case was that the python version was hard coded at 3.5; perhaps it would be better to replace the constant `PYTHON3_VERSION` with a calculated value that is based on the default system python? And would printing as the first line of output the version being used for static analysis be helpful?

**Comment by miedzinski** [comment]
`PYTHON3_VERSION` default is being discussed in #2836.

**Comment by JukkaL** [comment]
Yeah, the next mypy release will use 3.6 as the default target Python version, so this will get better.

I agree that the error messages could be better if you try to use something only available in a later Python version. Also, mypy could also tell if something is undefined because it's platform-specific thing and not defined for the current target platform.
